usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedAgency.java

```diff
@@ -0,0 +1,48 @@
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.ParsedEntity;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfsentity.Agency;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsDataRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsSpecRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+public class ProcessParsedAgency {
+    private final GtfsSpecRepository gtfsSpecRepository;
+    private final ValidationResultRepository resultRepository;
+    private GtfsDataRepository gtfsDataRepository;
+
+    public ProcessParsedAgency(GtfsSpecRepository gtfsSpecRepository, ValidationResultRepository resultRepository,
+                               GtfsDataRepository gtfsDataRepository) {
+        this.gtfsSpecRepository = gtfsSpecRepository;
+        this.resultRepository = resultRepository;
+        this.gtfsDataRepository = gtfsDataRepository;
+    }
+
+    public Agency.AgencyBuilder getBuilder(String agencyName, String agencyUrl, String agencyTimezone) {
```

Comment:
Not wrong but would have a preference for having the builder injected into the use case through a constructor parameter